src/rpc.ts

```diff
@@ -0,0 +1,157 @@
+import axios from 'axios';
+import * as _ from 'lodash';
+import { REQUEST_BATCH_SIZE } from './constants';
+
+export type RPCRequest = {
+  method: string;
+  params: any[];
+};
+
+export type RPCRequestRaw = RPCRequest & {
+  jsonrpc: string;
+  id: string;
+};
+
+export type RPCResponse = {
+  success: boolean;
+  result: any;
+};
+
+export type ProviderConfig = {
+  URL: string;
+  unsupportedMethods?: string[];
+  supportBatchRequests?: boolean;
+  batchSize?: number;
+};
+
+export class RPC {
+  constructor(protected provider: ProviderConfig) {}
+
+  async request(request: RPCRequest) {
+    if (
+      this.provider.unsupportedMethods &&
+      this.provider.unsupportedMethods.includes(request.method)
+    ) {
+      throw new Error('method not supported by the provider');
+    }
+    return await this._retryRequest(request);
+  }
+
+  async requestBatch(requests: RPCRequest[]) {
+    if (
+      this.provider.unsupportedMethods &&
+      requests.some(r => this.provider.unsupportedMethods!.includes(r.method))
+    ) {
+      throw new Error('method not supported by the provider');
+    }
+
+    if (this.provider.supportBatchRequests) {
+      const requestChunks = _.chunk(
+        requests,
+        this.provider.batchSize || REQUEST_BATCH_SIZE,
+      );
+      const res = [];
+      for (const chunk of requestChunks) {
+        const batchRes = await this._retryBatch(chunk);
+        res.push(batchRes);
+      }
+      return res.flat();
+    } else {
+      const res = [];
+      for (const request of requests) {
+        const r = await this._retryRequest(request);
+        res.push(r);
+      }
+      return res;
+    }
+  }
+
+  private async _retryRequest(
+    _request: RPCRequest,
+    retry: number = 5,
+  ): Promise<RPCResponse> {
+    const request = [
+      {
+        ..._request,
+        jsonrpc: '2.0',
+        id: Math.floor(Math.random() * 2 ** 64).toFixed(),
+      },
+    ];
+
+    for(let i = retry; i > 0; i--) {
```

Comment:
Space between `for` and `(`. Also let's make this `for` consistent with one on line 109.